docs/development/releasing.rst

```diff
@@ -12,231 +12,249 @@ more of the process, if not all.
-     will be appended"--it will actually be ".dev" without the 0).  For
-     example, if the just-released version was "0.1" the default next version
-     will be "0.2".  If we want the next version to be, say "0.1.1", or "1.0",
-     then that must be entered manually.
+#. Now go back and check out the tag of the released version with
+   ``git checkout v<version>``.  For example::
 
- 14. You will be shown a diff of CHANGES.rst showing that a new section has
-     been added for the new development version, and showing that the version
-     has been updated in setup.py.  Enter 'Y' to commit these changes.
+      $ git checkout v0.2.2
 
- 15. When asked to push the changes to a remote repository, enter 'Y'.  This
-     should complete the portion of the process that's automated at this point.
+   Don't forget to remove any non-commited files with::
 
- 16. Check out the tag of the released version.  For example::
+      $  git clean -dfx
 
-         $ git checkout v0.1
+#. Create the source distribution by doing::
 
- 17. Create the source distribution by doing::
-
-         $ git clean -dfx  # just in case
          $ python setup.py build sdist
 
-     Copy the produced ``.tar.gz`` somewhere and verify that you can unpack it,
-     build it, and get all the tests to pass.  It would be best to create a new
-     virtualenv in which to do this.
-
      .. note::
 
-         In the future, the ``build`` command should run automatically as a
+         In the future, the ``build`` command may run automatically as a
          prerequisite for ``sdist``.  But for now, make sure to run it
          whenever running ``sdist`` to ensure that all Cython sources and
          other generated files are built.
 
- 18. Register the release on PyPI with::
+#. Run the tests in an environment that mocks up a "typical user" scenario.
+   This is not strictly necessary because you ran the tests above, but
+   it can sometimes be useful to catch subtle bugs that might come from you
+   using a customized developer environment.  For more on setting up virtual
+   environments, see :ref:`virtual_envs`, but for the sake of example we will
+   assume you're using `Anaconda`_. Do::
 
-         $ python setup.py register
+      $ conda create -n astropy_release_test_v<verson> numpy
+      $ source activate astropy_release_test_v<verson>
+      $ cd dist
+      $ mkdir test
+      $ pip install astropy-<version>.tar.gz
+      $ python -c 'import astropy; astropy.test(remote_data=True)'
+      $ source deactivate
+      $ cd <back to the source directory>
 
- 19. Upload the source distribution to PyPI; this is preceded by re-running
-     the sdist command, which is necessary for the upload command to know
-     which distribution to upload::
+   Assuming everything went smoothly with the last step, we are ready to proceed
+   with the release.  If not, you'll have to back up to before starting this
+   procedure and start over when things are fixed.
```

Comment:
I would also add a paragraph about the case where things don't go smoothly - in this case I usually remove the last two commits and then fix issues, and then start the release process again.